gen/internal/tests/err_on_unknown_enum_value/err_on_unknown_enum_value.go

```diff
@@ -0,0 +1,211 @@
+func (v EnumDefault) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	enc.AddInt32("value", int32(v))
+	switch int32(v) {
+	case 0:
+		enc.AddString("name", "Foo")
+	case 1:
+		enc.AddString("name", "Bar")
+	case 2:
+		enc.AddString("name", "Baz")
+	}
+	return nil
+}
+
+// Ptr returns a pointer to this enum value.
+func (v EnumDefault) Ptr() *EnumDefault {
+	return &v
+}
+
+// ToWire translates EnumDefault into a Thrift-level intermediate
+// representation. This intermediate representation may be serialized
+// into bytes using a ThriftRW protocol implementation.
+//
+// Enums are represented as 32-bit integers over the wire.
+func (v EnumDefault) ToWire() (wire.Value, error) {
+	return wire.NewValueI32(int32(v)), nil
+}
+
+// FromWire deserializes EnumDefault from its Thrift-level
+// representation.
+//
+//   x, err := binaryProtocol.Decode(reader, wire.TI32)
+//   if err != nil {
+//     return EnumDefault(0), err
+//   }
+//
+//   var v EnumDefault
+//   if err := v.FromWire(x); err != nil {
+//     return EnumDefault(0), err
+//   }
+//   return v, nil
+func (v *EnumDefault) FromWire(w wire.Value) error {
+	*v = (EnumDefault)(w.GetI32())
+	return nil
+}
+
+// String returns a readable string representation of EnumDefault.
+func (v EnumDefault) String() string {
+	w := int32(v)
+	switch w {
+	case 0:
+		return "Foo"
+	case 1:
+		return "Bar"
+	case 2:
+		return "Baz"
+	}
+	return fmt.Sprintf("EnumDefault(%d)", w)
+}
+
+// Equals returns true if this EnumDefault value matches the provided
+// value.
+func (v EnumDefault) Equals(rhs EnumDefault) bool {
+	return v == rhs
+}
+
+// MarshalJSON serializes EnumDefault into JSON.
+//
+// If the enum value is recognized, its name is returned. Otherwise,
+// its integer value is returned.
+//
+// This implements json.Marshaler.
+func (v EnumDefault) MarshalJSON() ([]byte, error) {
+	switch int32(v) {
+	case 0:
+		return ([]byte)("\"Foo\""), nil
+	case 1:
+		return ([]byte)("\"Bar\""), nil
+	case 2:
+		return ([]byte)("\"Baz\""), nil
+	}
+	return nil, errors.New(fmt.Sprintf("unknown value for enum %s", v))
+}
+
+// UnmarshalJSON attempts to decode EnumDefault from its JSON
+// representation.
+//
+// This implementation supports both, numeric and string inputs. If a
+// string is provided, it must be a known enum name.
+//
+// This implements json.Unmarshaler.
+func (v *EnumDefault) UnmarshalJSON(text []byte) error {
+	d := json.NewDecoder(bytes.NewReader(text))
+	d.UseNumber()
+	t, err := d.Token()
+	if err != nil {
+		return err
+	}
+
+	switch w := t.(type) {
+	case json.Number:
+		x, err := w.Int64()
+		if err != nil {
+			return err
+		}
+		if x > math.MaxInt32 {
+			return fmt.Errorf("enum overflow from JSON %q for %q", text, "EnumDefault")
+		}
+		if x < math.MinInt32 {
+			return fmt.Errorf("enum underflow from JSON %q for %q", text, "EnumDefault")
+		}
+		*v = (EnumDefault)(x)
+		return nil
+	case string:
+		return v.UnmarshalText([]byte(w))
```

Comment:
I think it makes sense to also throw an error during unmarshal of unknown values. The flag is not specific to marshaling (and probably shouldn't be).